url71665d7d68_branscha_lib_scripty_tgz-pJ8-com_sdicons_scripty_ScriptyAutoReplJ8/src/com/sdicons/scripty/spec/args/ArgListBuilderUtil.java

```diff
@@ -0,0 +1,714 @@
+        {
+            try
+            {
+                NOARGS.guard(aArgs, aCtx);
+                return SHORT_TUPLE;
+            }
+            catch (ArgSpecException e)
+            {
+                throw new CommandException(e.getMessage());
+            }
+        }
+
+        @ScriptyCommand(name="String")
+        public static TypeTuple stringType(Object[] aArgs, IContext aCtx)
+        throws CommandException
+        {
+            try
+            {
+                NOARGS.guard(aArgs, aCtx);
+                return STRING_TUPLE;
+            }
+            catch (ArgSpecException e)
+            {
+                throw new CommandException(e.getMessage());
+            }
+        }
+
+        // type="OneOf (Integer) (Long) (BigDecimal)"
+        private static VarArgList ONEOF_ARGLIST = new VarArgList(new FixedArg[]{}, new VarArg(SPEC_TYPE), 2, -1, new NamedArg[]{});
+
+        @ScriptyCommand(name="OneOf")
+        public static TypeTuple oneOfType(IContext aCtx, Object[] aArgs)
+        throws CommandException
+        {
+            try
+            {
+                ONEOF_ARGLIST.guard(aArgs, aCtx);
+                Map<String, IArgMapping> lMappings = new HashMap<String, IArgMapping>();
+                ITypeSpec[] lSpecs = new ITypeSpec[aArgs.length-1];
+                for(int i = 1; i < aArgs.length; i++)
+                {
+                    TypeTuple lTuple = (TypeTuple) aArgs[i];
+                    lSpecs[i-1] = lTuple.getX();
+                    if(lTuple.getY() != null) lMappings.putAll(lTuple.getY());
+                }
+                return new TypeTuple(new OrType(lSpecs), lMappings);
+            }
+            catch (ArgSpecException e)
+            {
+                throw new CommandException(e.getMessage());
+            }
+        }
+
+        // type="Custom com.sdicons.MySpec"
+        private static StdArgList CUSTOM_ARGLIST = new StdArgList(new FixedArg[]{FIXED_CLASSNAME}, new OptionalArg[]{}, new NamedArg[]{});
+
+        @ScriptyCommand(name="Custom")
+        public static TypeTuple customSpec(IContext aCtx, Object[] aArgs)
+        throws CommandException
+        {
+            try
+            {
+                aArgs = CUSTOM_ARGLIST.guard(aArgs, aCtx);
+                String lClassName = (String) aArgs[1];
+
+                try
+                {
+                    Class lClass = Class.forName(lClassName);
+                    if(ITypeSpec.class.isAssignableFrom(lClass))
+                    {
+                        ITypeSpec lSpec = (ITypeSpec) lClass.newInstance();
+                        return new TypeTuple(lSpec, null);
+                    }
+                    else
+                    {
+                        String lMsg = String.format("The class '%s' does not seem to implement ITypeSpec.", lClassName);
+                        throw new CommandException(lMsg);
+                    }
+                }
+                catch (ClassNotFoundException e)
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: While wrapping the caught exception into a custom one, information about the caught exception is being lost, including information about the stack trace of the exception.

**Fix**: If the caught exception object does not contain sensitive information, consider passing it as the &#34;rootCause&#34; or inner exception parameter to the constructor of the new exception before throwing the new exception. (Note that not all exception constructors support inner exceptions. Use a wrapper exception that supports inner exceptions.) 
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)
Similar issue at line numbers 412, 417, 423.